user-console/src/main/resources/org/pentaho/mantle/public/browser/js/browser.js

```diff
@@ -500,36 +516,65 @@ define([
 
     updateData: function () {
       var myself = this;
+
+      const trashFolder = {
+        "file": {
+          "trash": "trash",
+          "createdDate": "1365427106132",
+          "fileSize": "0",
+          "folder": true,
+          "hidden": "false",
+          "objectId:": jQuery.i18n.prop('trash'),
+          "locale": "en",
+          "locked": "false",
+          "name": jQuery.i18n.prop('trash'),
+          "ownerType": "-1",
+          "path": ".trash",
+          "title": jQuery.i18n.prop('trash'),
+          "versioned": "false"
+        }
+      };
+
       myself.set("runSpinner", true);
-      myself.fetchData("/", function (response) {
-        var trash = {
-          "file": {
-            "trash": "trash",
-            "createdDate": "1365427106132",
-            "fileSize": "0",
-            "folder": "true",
-            "hidden": "false",
-            "id:": jQuery.i18n.prop('trash'),
-            "locale": "en",
-            "locked": "false",
-            "name": jQuery.i18n.prop('trash'),
-            "ownerType": "-1",
-            "path": ".trash",
-            "title": jQuery.i18n.prop('trash'),
-            "versioned": "false"
+      myself.clearTreeCache();
+      myself.fetchTreeRootData(function (response) {
+        //Add trash folder once to the first Repository folder
+        //This should only ever happen once anyway, but implemented safely
+        for( let i=0; i < response.children.length; i++){
+          let childFolder = response.children[0];
+          if (_isRepositoryPath(childFolder.file.path)){
+            childFolder.children.push(trashFolder);
+            break;
```

Comment:
Just realized that this will likely not work properly on a CE server, where only the repository provider is installed. The children will be `home`, `public`.
Would better have a function to "find" the "Repository" root folder on the given response and then, if found, would add the trash to its children.

Reply:
That works. But please create a function such as "get repository parent folder" to contain the code to find the repository root. And, also use the new `isRepositoryPath()` to detect if it is the repository.

Reply:
ok, I can include `isRepositoryPath()` but I want to be sure that the path is _only_ "/" as the function just checks that the first character is "/"

Reply:
I need to check that the path is == "/" and not just "starts with" as `isRepositoryPath()` does

Reply:
Oh, yes, that's a good point. Although, I'd be happy if you'd commit that sin, of assuming that the first repo thing found is good enough. I mean, I prefer that to repeating the logic.
However, you could easily create a `ROOT_PATH` constant you could share between the existing method, `isRepositoryPath` and another one you'd create, such as `isRepositoryRootPath`.

Reply:
If I just did `isRepositoryPath`, for CE, it would put the trash folder under the home folder. I can make a constant and a new function if you would like.

Reply:
Oh, good point!